spikeextractors/extractors/neuroscopeextractors/neuroscopeextractors.py

```diff
@@ -0,0 +1,451 @@
+    installed = True  # check at class level if installed or not
+    is_writable = True
+    mode = 'folder'
+    installation_mesg = 'please install bs4 and lxml to use this extractor'  # error message when not installed
+            
+    def __init__(self, folder_path, subset_channels=None):
+        assert HAVE_BS4_LXML, self.installation_mesg
+        RecordingExtractor.__init__(self)
+        self._recording_file = folder_path
+        
+        fpath_base, fname = os.path.split(folder_path)
+        xml_filepath = os.path.join(folder_path, fname + '.xml')
+        dat_filepath = os.path.join(folder_path, fname + '.dat')
+        
+        with open(xml_filepath, 'r') as xml_file:
+            contents = xml_file.read()
+            soup = BeautifulSoup(contents, 'lxml')
+            # Normally, this would be a .xml, but there were strange issues
+            # in the write_recording method that require it to be a .lxml instead
+            # which also requires all capital letters to be removed from the tag names
+        
+        n_bits = int(soup.nbits.string)
+        dtype='int'+str(n_bits)
+        num_channels = int(soup.nchannels.string)
+        
+        if subset_channels is not None:
+            num_channels = len(subset_channels)
+        
+        sampling_frequency = float(soup.samplingrate.string)
+        
+        BinDatRecordingExtractor.__init__(self, dat_filepath, sampling_frequency=sampling_frequency,
+                                          dtype=dtype, numchan=num_channels)
+        
+        self._kwargs = {'folder_path': str(Path(folder_path).absolute()),
+                        'subset_channels': subset_channels}
+        
+        
+    @staticmethod
+    def write_recording(recording, save_path, dtype='int32'):
+        """ Convert and save the recording extractor to Neuroscope format
+
+        parameters
+        ----------
+        recording: RecordingExtractor
+            The recording extractor to be converted and saved
+        save_path: str
+            Full path to desired target folder
+        dtype: str
+            Data type to be used in writing. Will throw a warning if stored recording type from get_traces() does not match.
+        """
+        _, RECORDING_NAME = os.path.split(save_path)
+        save_xml = "{}/{}.xml".format(save_path,RECORDING_NAME)
+
+        # write recording
+        recording_fn = os.path.join(save_path, RECORDING_NAME)
+
+        # create parameters file if none exists
+        if not os.path.isfile(save_xml):
+            soup = BeautifulSoup("",'xml')
+
+            new_tag = soup.new_tag('nbits')
+            dtype = recording.get_dtype()
+            
+            if not any([dtype == x for x in ['int16', 'int32']]):
+                print('Warning: Unsupported data type passed (',dtype,'); converting to default of int32!',sep="")
+                dtype = 'int32'
+                
+            n_bits = str(dtype)[3:5]
```

Comment:
This will fail if one passes a `numpy` dtype. I see in line 97 that you only accept strings, so it's fine here. In general, a more robust way is to find the `int` substring and use relative indexes from that position :)

Reply:
Great point; I've implemented the robust way now!